server/authorization.go

```diff
@@ -433,17 +435,27 @@ func addRequestHeaderTransforms(req *authv3.CheckRequest, envRequest *config.Env
 				targetPath = targetPath + "?" + strings.Join(queryParams, "&")
 			}
 
-			addRequestHeader(okResponse, envoyPathHeader, targetPath, false)
-
 			// header transforms
-			for _, t := range transforms.HeaderTransforms.Remove {
-				t = strings.ToLower(t)
-				for hdr := range req.Attributes.Request.Http.Headers {
-					if util.SimpleGlobMatch(t, hdr) {
-						okResponse.HeadersToRemove = append(okResponse.HeadersToRemove, hdr)
+			for hdr := range req.GetAttributes().GetRequest().GetHttp().GetHeaders() {
+				// strip all x-apigee-* request headers before target
+				if strings.HasPrefix(hdr, "x-apigee-") {
```

Comment:
We are currently relying on the `x-apigee-message-id` request header in the access log entry to generate the same response header. Have we confirmed that removing this here will not make that header absent from the log entry?

Reply:
The directives to add response headers are based on the incoming request from ingress. These directives to remove headers only affects the outgoing request to the target - and only after all ARC processing is complete.